[CI] IndexLifecycleExplainResponseTests testConcurrentToXContent failing

**Build Scans:**
- [elasticsearch-periodic #4733 / openjdk18_java-matrix](https://gradle-enterprise.elastic.co/s/fr4vjafzkijn6)
- [elasticsearch-periodic-platform-support #4663 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/yqjfp3bqg72i4)

**Reproduction Line:**
```
./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.ilm.IndexLifecycleExplainResponseTests.testConcurrentToXContent" -Dtests.seed=B9AEC42CB80C6EB5 -Dtests.locale=lt -Dtests.timezone=ACT -Druntime.java=18
```

**Applicable branches:**
7.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.core.ilm.IndexLifecycleExplainResponseTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testConcurrentToXContent),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failed to round trip over human readable YAML
Canonical is:
{
  "index" : "tQAbwLnOir",
  "managed" : true,
  "policy" : "KSJKdIBAns",
  "lifecycle_date" : "2001-09-11T14:47:43.214Z",
  "lifecycle_date_millis" : 1000219663214,
  "age" : "8455.64d",
  "phase" : "ekRprWEdVy",
  "phase_time" : "65465859-07-22T00:37:02.263Z",
  "phase_time_millis" : 2065840820354222263,
  "action" : "NlrlgDhscg",
  "action_time" : "83056496-11-03T23:23:55.176Z",
  "action_time_millis" : 2620947717012235176,
  "step" : "QmUJJUIAOS",
  "step_time" : "276645879-12-21T01:43:53.912Z",
  "step_time_millis" : 8730038588003033912,
  "is_auto_retryable_error" : false,
  "failed_step_retry_count" : 5,
  "repository_name" : "yjuTVEzykY",
  "snapshot_name" : "tCuIlYAOpM",
  "shrink_index_name" : "wYgUJvULLa"
}
Wanted : ---
index: "tQAbwLnOir"
managed: true
policy: "KSJKdIBAns"
lifecycle_date: "2001-09-11T14:47:43.214Z"
lifecycle_date_millis: 1000219663214
age: "8455.63d"
phase: "ekRprWEdVy"
phase_time: "65465859-07-22T00:37:02.263Z"
phase_time_millis: 2065840820354222263
action: "NlrlgDhscg"
action_time: "83056496-11-03T23:23:55.176Z"
action_time_millis: 2620947717012235176
step: "QmUJJUIAOS"
step_time: "276645879-12-21T01:43:53.912Z"
step_time_millis: 8730038588003033912
is_auto_retryable_error: false
failed_step_retry_count: 5
repository_name: "yjuTVEzykY"
snapshot_name: "tCuIlYAOpM"
shrink_index_name: "wYgUJvULLa"

But got: ---
index: "tQAbwLnOir"
managed: true
policy: "KSJKdIBAns"
lifecycle_date: "2001-09-11T14:47:43.214Z"
lifecycle_date_millis: 1000219663214
age: "8455.64d"
phase: "ekRprWEdVy"
phase_time: "65465859-07-22T00:37:02.263Z"
phase_time_millis: 2065840820354222263
action: "NlrlgDhscg"
action_time: "83056496-11-03T23:23:55.176Z"
action_time_millis: 2620947717012235176
step: "QmUJJUIAOS"
step_time: "276645879-12-21T01:43:53.912Z"
step_time_millis: 8730038588003033912
is_auto_retryable_error: false
failed_step_retry_count: 5
repository_name: "yjuTVEzykY"
snapshot_name: "tCuIlYAOpM"
shrink_index_name: "wYgUJvULLa"

```

**Issue Reasons:**
- [7.17] 2 failures in test testConcurrentToXContent (0.6% fail rate in 317 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponse.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponse.java
@@ -23,6 +23,7 @@ import org.elasticsearch.xcontent.json.JsonXContent;
 
 import java.io.IOException;
 import java.util.Objects;
+import java.util.function.Supplier;
 import java.util.stream.Stream;
 
 public class IndexLifecycleExplainResponse implements ToXContentObject, Writeable {
@@ -123,6 +124,8 @@ public class IndexLifecycleExplainResponse implements ToXContentObject, Writeabl
     private final String repositoryName;
     private final String snapshotName;
     private final String shrinkIndexName;
+
+    Supplier<Long> nowSupplier = System::currentTimeMillis; // Can be changed for testing
 
     public static IndexLifecycleExplainResponse newManagedIndexResponse(
         String index,
@@ -412,11 +415,11 @@ public class IndexLifecycleExplainResponse implements ToXContentObject, Writeabl
         return failedStepRetryCount;
     }
 
-    public TimeValue getAge() {
+    public TimeValue getAge(Supplier<Long> now) {
         if (lifecycleDate == null) {
             return TimeValue.MINUS_ONE;
         } else {
-            return TimeValue.timeValueMillis(Math.max(0L, System.currentTimeMillis() - lifecycleDate));
+            return TimeValue.timeValueMillis(Math.max(0L, now.get() - lifecycleDate));
         }
     }
 
@@ -441,7 +444,7 @@ public class IndexLifecycleExplainResponse implements ToXContentObject, Writeabl
             builder.field(POLICY_NAME_FIELD.getPreferredName(), policyName);
             if (lifecycleDate != null) {
                 builder.timeField(LIFECYCLE_DATE_MILLIS_FIELD.getPreferredName(), LIFECYCLE_DATE_FIELD.getPreferredName(), lifecycleDate);
-                builder.field(AGE_FIELD.getPreferredName(), getAge().toHumanReadableString(2));
+                builder.field(AGE_FIELD.getPreferredName(), getAge(nowSupplier).toHumanReadableString(2));
             }
             if (phase != null) {
                 builder.field(PHASE_FIELD.getPreferredName(), phase);

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponseTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponseTests.java
@@ -121,7 +121,7 @@ public class IndexLifecycleExplainResponseTests extends AbstractSerializingTestC
             null,
             null
         );
-        assertThat(response.getAge(), equalTo(TimeValue.ZERO));
+        assertThat(response.getAge(System::currentTimeMillis), equalTo(TimeValue.ZERO));
     }
 
     @Override

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponseTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponseTests.java
@@ -34,11 +34,16 @@ import static org.hamcrest.Matchers.startsWith;
 public class IndexLifecycleExplainResponseTests extends AbstractSerializingTestCase<IndexLifecycleExplainResponse> {
 
     static IndexLifecycleExplainResponse randomIndexExplainResponse() {
+        final IndexLifecycleExplainResponse indexLifecycleExplainResponse;
         if (frequently()) {
-            return randomManagedIndexExplainResponse();
+            indexLifecycleExplainResponse = randomManagedIndexExplainResponse();
         } else {
-            return randomUnmanagedIndexExplainResponse();
+            indexLifecycleExplainResponse = randomUnmanagedIndexExplainResponse();
         }
+        long now = System.currentTimeMillis();
+        // So that now is the same for the duration of the test. See #84352
+        indexLifecycleExplainResponse.nowSupplier = () -> now;
+        return indexLifecycleExplainResponse;
     }
 
     private static IndexLifecycleExplainResponse randomUnmanagedIndexExplainResponse() {

